src/SplitText.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+export const SplitText = ({ children, className, separator, ...rest }) => {
+    const words = children.split(separator);
+    const lastChar = children[children.length - 1];
+
+    return words.map((word, index) => {
+        const withSeparator = index !== words.length - 1 || lastChar === separator;
```

Comment:
Sorry for not being clear enough 😅
The problem with just checking the last char is if the separator is more than one char.

Maybe we should just do something like:
```suggestion
        const withSeparator = index !== words.length - 1 || word === "";
```

Because of cases like this, right?

![imagem](https://user-images.githubusercontent.com/11398440/80209887-ba442d00-862a-11ea-875f-90e27a19db7f.png)

Reply:
> Maybe we should just do something like:

That doesn't work either, you can see by splitting an empty space:

![clg](https://user-images.githubusercontent.com/33324750/80224567-c8517800-8641-11ea-99a3-e5dd289c4f5c.png)

As split returns two empty strings, the separator would be true for both so we would have two empty spaces. 

Nonetheless, that pointed me in the right direction and I got a working solution: 

```js
return words.map((word, index) => {
    const withWord = word !== '';
    const withSeparator = index !== words.length - 1;
    const shouldRender = withWord || withSeparator;

    return (
        shouldRender && <span
            key={ `word-${index}` }
            className={ classNames('react-split-text', className) }
            { ...rest }>
            { withWord && word }
            { withSeparator && separator }
        </span>
    );
});
```

A quick rundown:

- We always add a separator unless we're in the last position, in which we render nothing because that separator was rendered in the previous position already;
- That `shouldRender` check is to ignore the last position we talked about in the previous point, or else a `<span>` with nothing in it would be rendered and we don't want that (that's the little green point in my [previous comment](https://github.com/moxystudio/react-split-text/pull/1#discussion_r414437673));

Check it out:

![split](https://user-images.githubusercontent.com/33324750/80223997-013d1d00-8641-11ea-822e-0b0a31ea9fe6.png)